# Integration of T Cell Repertoire, CyTOF, genotyping and symptomatology data reveals subphenotypic variability in COVID-19 Patients

## Abstract
COVID-19 manifests across a spectrum from asymptomatic to fatal outcomes. While various studies have proposed biomarkers, integrated analysis combining mass cytometry (CyTOF) and T-cell receptor sequencing (TCRseq) with genotypic and symptomatic data remains underexplored. We applied the Latent Class Model with Bayesian Information Criterion (LCM-BIC) to integrate immunophenotyping data, including monocyte subpopulations from CyTOF and TCRseq, with symptomatic and single-nucleotide polymorphisms data from ACE2 (rs2285666), MX1 (rs469390), and TMPRSS2 (rs2070788) in 61 Spanish COVID-19 patients (33 mild, 28 severe). Deep learning techniques were used to analyze TCR Î²-chain Complementarity Determining Region 3 (CDR3) amino acid sequences. LCM-BIC identified three distinct patient clusters, characterized by 70 variables selected by BIC. Cluster 1 was predominantly associated with severe cases. Significant differences in key variables such as monocyte subpopulations, TCR diversity, V allele usage, and sequence motifs were observed between clusters. Deep learning analysis of CDR3 sequences allowed precise distinction of Cluster 1, highlighting unique immunological SARS-CoV-2 TCR signatures linked to disease severity. This study demonstrates the utility of integrating advanced immunophenotyping and genetic profiling to improve understanding of COVID-19 severity, enabling the identification of novel immunological signatures and enhancing patient stratification based on their immunological profile.

## Graphical Abstract

![](https://github.com/fmarinb/COVID19_LCM_integration/blob/main/GA.jpeg) 


## Repertorie data avaliability

The 173 bulk TCR-beta chain pre-processed files are stored at the European Genome Phenome Archive (EGA) upon formal request (https://ega-archive.org/datasets/EGAD50000000477).
